lib/AST/ASTPrinter.cpp

```diff
@@ -2518,6 +2518,11 @@ void PrintAST::printInherited(const Decl *decl) {
   Printer << ": ";
 
   interleave(TypesToPrint, [&](InheritedEntry inherited) {
+    // Printing a SuppressedTypeRepr will add the '~', so only print it
+    // if we will not be using the Repr's print method for this type.
+    if (!Options.PreferTypeRepr && inherited.isSuppressed)
+      Printer << "~";
```

Comment:
We'll need to emit a feature guard when printing these suppressed constraints in swiftinterfaces in order to avoid confails with older compilers that don't understand the new operator.

Reply:
Good catch. Despite the only valid uses of them today being guarded by `$MoveOnly`, previous compilers that understood `$MoveOnly` did not also understand the `~` syntax.